src/main/java/com/redhat/devtools/lsp4ij/server/definition/launching/UserDefinedFormattingFeature.java

```diff
@@ -78,10 +79,12 @@ public boolean isFormatOnStatementTerminator(@NotNull PsiFile file) {
     }
 
     @Override
-    @Nullable
+    @NotNull
     public String getFormatOnStatementTerminatorCharacters(@NotNull PsiFile file) {
         ClientSideOnTypeFormattingSettings settings = getClientSideOnTypeFormattingSettings();
-        return settings != null ? settings.formatOnStatementTerminatorCharacters : super.getFormatOnStatementTerminatorCharacters(file);
+        return (settings != null) && StringUtil.isNotEmpty(settings.formatOnStatementTerminatorCharacters) ?
```

Comment:
If the subset of format-on-statement terminator characters is defined, use it; otherwise use the language grammar-level statement terminator characters.